### DISCUSSION TITLE
Steps to run local dev server

### DISCUSSION THREAD

**Discussion by multivoltage** [resend/react-email#1814] [Q&A] [answered]
After I opened a ticket for a new idea now I tried to implement the idea.

Steps:
1. git clone "repo"
2. use node 18 (since all packages has engine >= 18
3. pnpm install
4. pnpm run dev

After 1 minute server listening on localhost:3000 and I can see the templates.

When I change some code inside "packages/react-emails" nothing happends.
When I put broken code inside "packages/react-emails" nothing happends.

I read also https://github.com/resend/react-email/issues/483

**Comment by gabrielmfern** [comment, marked as answer]
First things first, the discussion you want to implement needs a proper path of implementation where we agree with, so you don't waste any time making a PR that isn't going to be merged. And, as things were left there, you haven't convinced us that what we already have isn't enough.

Secondly, the dev command is for development of the demo emails, not for preview server development. To run an equivalent of `next dev` for the preview server inside the demo, you can run `pnpm tsx ../../packages/react-email/src/cli/index.ts dev` inside `apps/demo` and it will work as you are expecting.

**Reply by multivoltage** [reply]
I opened a ticket for "add dynamic router" like next style (example [slug].tsx). The idea was blocked in favour of something like "editing params in storybook style".
I agree with that. Since my idea proposal Is very different than the suggestion I think that open a new ticket for an idea of "storybook like" Is better than keep going in the original ticket.
If not I' ll wrote that.

I Will tru with your suggestion for run dev
